qemu/tests/usb_host.py

```diff
@@ -52,62 +109,79 @@ def usb_dev_unplug():
-    lsusb_cmd = "lsusb -v -d %s" % device
-    monitor_add = "device_add usb-host,bus=usbtest.0,id=usbhostdev"
-    monitor_add += ",vendorid=0x%s" % vendorid
-    monitor_add += ",productid=0x%s" % productid
-    monitor_del = "device_del usbhostdev"
+    lsusb_cmd = "lsusb -v -d %s:%s" % (vendorid, productid)
     match_add = "New USB device found, "
     match_add += "idVendor=%s, idProduct=%s" % (vendorid, productid)
     match_del = "USB disconnect"
-
-    error_context.context("Check usb device %s on host" % device, logging.info)
-    try:
-        process.system(lsusb_cmd, shell=True)
-    except:
-        test.cancel("Device %s not present on host" % device)
+    usb_stick = "Mass Storage" in process.getoutput(lsusb_cmd)
 
     error_context.context("Log into guest", logging.info)
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
     session = vm.wait_for_login()
 
-    repeat_times = int(params.get("usb_repeat_times", "1"))
-    for i in range(repeat_times):
-        if params.get("usb_check_isobufs", "no") == "no":
-            error_context.context("Hotplug (iteration %i)" % (i + 1),
-                                  logging.info)
-        else:
-            # The value of isobufs could only be in '4, 8, 16'
-            isobufs = (2 << (i % 3 + 1))
-            monitor_add = "device_add usb-host,bus=usbtest.0,id=usbhostdev"
-            monitor_add += ",vendorid=0x%s" % vendorid
-            monitor_add += ",productid=0x%s" % productid
-            monitor_add += ",isobufs=%d" % isobufs
-            error_context.context("Hotplug (iteration %i), with 'isobufs'"
-                                  " option set to %d" % ((i + 1), isobufs),
-                                  logging.info)
-        usb_dev_hotplug()
+    try:
         usb_dev_verify()
-        usb_dev_unplug()
+        if usb_stick:
+            iozone_test = None
+            mount_point = _get_usb_mount_point()
+            iozone_test = generate_instance(params, vm, 'iozone')
+            _usb_stick_io(mount_point)
+        usb_devs = get_usb_host_dev()
+        for dev in usb_devs:
+            usb_dev_unplug(dev)
 
-    session.close()
+        repeat_times = int(params.get("usb_repeat_times", "1"))
+        for i in range(repeat_times):
+            msg = "Hotplug (iteration %d)" % (i+1)
+            usb_params["id"] = "usbhostdev%s" % i
+            if params.get("usb_check_isobufs", "no") == "yes":
+                # The value of isobufs could only be in '4, 8, 16'
+                isobufs = (2 << (i % 3 + 1))
+                usb_params["isobufs"] = isobufs
+                msg += ", with 'isobufs' option set to %d." % isobufs
+            error_context.context(msg, logging.info)
+            usb_dev = qdevices.QDevice("usb-host", usb_params)
+            usb_dev_hotplug(usb_dev)
+            usb_dev_verify()
+            if usb_stick:
+                mount_point = _get_usb_mount_point()
+                _usb_stick_io(mount_point, bg=True)
```

Comment:
Avoid the trace back be shown in debug log is better.

Reply:
@ybduan  As we discussed, this can be implement by update `_usb_stick_io`, refer:
```
        if bg:
            iozone_cmd = params.get("iozone_cmd_bg", " -az -I -g 1g -f %s")
            bg = utils_test.BackgroundTest(iozone_test.run,
                                           iozone_cmd % testfile)
            bg.start()
            if not utils_misc.wait_for(bg.is_alive, timeout=10):
                test.fail("*****")  ---> need param "test"
```

Reply:
Updated, thanks!